src/rest/applications/mcm/resubmission.py

```diff
@@ -0,0 +1,428 @@
+        self, chain_request_prepid: str, root_request_processing: RootRequestReset
+    ) -> None:
+        """
+        Injects a chained request based on how its root request was reset.
+        This MUST BE executed only after rewinding a chained request to root and reserving it again.
+
+        Args:
+            chain_request_prepid: Chained request identifier.
+            root_request_processing: Describes how the root request was reset.
+        """
+        chained_request: Union[dict, None] = self._mcm.get(
+            object_type="chained_requests", object_id=chain_request_prepid
+        )
+        if not chained_request:
+            raise ValueError(f"Chain request not found: {chain_request_prepid}")
+
+        if root_request_processing == RootRequestReset.KEEPS_OUTPUT:
+            # Just flow the chained request
+            flow_result = self._mcm.flow(chained_request_prepid=chain_request_prepid)
+            if not flow_result:
+                msg = f"Unable to flow the following chained request: {chain_request_prepid}"
+                self.logger.error(msg)
+                raise RuntimeError(msg)
+
+        elif root_request_processing in (
+            RootRequestReset.SOFT_RESET,
+            RootRequestReset.FULL_RESET,
+        ):
+            # Retrieve the newly created requests and
+            # make sure their approval/state is approve/approved
+            requests_in_chain = chained_request.get("chain", [])
+            to_approve = requests_in_chain[1:]
+            root_request_in_chain = requests_in_chain[0]
+            self.logger.info("Making sure non-root requests are approve/approved")
+            for req_prepid in to_approve:
+                self._approve_request_until(
+                    request_prepid=req_prepid,
+                    approval="approve",
+                    status="approved",
+                )
+
+            # Operate the root request in the chain and
+            # make sure its state is submit/submitted.
+            # INFO: This takes time ~3 to 5 min.
+            self.logger.info("Injecting chained request")
+            self._approve_request_until(
+                request_prepid=root_request_in_chain,
+                approval="submit",
+                status="submitted",
+            )
+
+        else:
+            raise NotImplementedError(
+                "There's no way to inject a chained request based on: %s",
+                root_request_processing,
+            )
+
+    def resubmit_chain_request(
+        self,
+        root_request_prepid: str,
+        datatier: str = "nanoaod",
+        soft_datatier: bool = False,
+        tracking_tag: Union[str, None] = None,
+    ) -> None:
+        """
+        Resubmit all the chained requests that reference a particular root request.
+        To achieve this, this procedure rewinds all the chained requests to the root step,
+        deleting and invalidating all the found requests except for the root one.
```

Comment:
This may not be optimal. Say only the Nano step has relevant changes, we could then only rewind to Mini and reuse the stored MiniAOD. Maybe add a note about it

Reply:
I included a description of the use case in the class [documentation](https://github.com/cms-PdmV/mcm_scripts/blob/70d710f49a8d2dd5b5a7be6111311f124d96879f/src/rest/applications/mcm/resubmission.py#L37-L42). Do you think it is a good idea to include it also here?